mssqlcli/telemetry.py

```diff
@@ -170,23 +173,33 @@ def _get_mssql_cli_version():
 def _get_user_id():
     config_dir = config.config_location()
     full_path = os.path.join(config_dir, MSSQL_CLI_TELEMETRY_ID_FILE)
-
-    if os.path.exists(full_path):
-        with open(full_path, 'r') as file:
-            user_id = file.read()
-            return user_id
-    else:
+    if _user_id_file_is_old(full_path) or not os.path.exists(full_path):
         with open(full_path, 'w') as file:
             user_id = _generate_user_id()
             file.write(user_id)
             return user_id
+    else:
+        with open(full_path, 'r') as file:
+            user_id = file.read()
+            return user_id
+
+
+def _user_id_file_is_old(id_file_path):
+    if os.path.exists(id_file_path):
+        last_24_hours = temp.now() - timedelta(hours=24)
+        id_file_creation_time = temp.fromtimestamp(os.path.getctime(id_file_path))
+
+        return id_file_creation_time < last_24_hours
+    return False
 
 
 @decorators.suppress_all_exceptions(fallback_return='')
 @decorators.hash256_result
 def _generate_user_id():
-    random_id = uuid.uuid4().hex
-    salt = uuid.uuid4().hex
+    random_id = binascii.hexlify(os.urandom(16)).decode() \

```

Comment:
We can combine this into one call to os.urandom(32) since the id/salt distinction doesn't really apply anymore?

Reply:
good point, will do.